gui/src/main/daemon-rpc.ts

```diff
@@ -551,12 +551,14 @@ export class DaemonRpc {
     listener.subscriptionId = subscriptionId;
   }
 
-  public async unsubscribeDaemonEventListener(
+  public unsubscribeDaemonEventListener(
     listener: SubscriptionListener<DaemonEvent>,
   ): Promise<void> {
     if (listener.subscriptionId) {
       return this.transport.unsubscribe('daemon_event', listener.subscriptionId);
     }
+
+    return Promise.resolve();
```

Comment:
FYI this is exactly what `async` keyword does. It also unwraps the `Promise` if you return one from `async` function, i.e if we were to translate the effect of `async` keyword into code I think it looks something like this:

```
function somethingAsync() {
  return new Promise((resolve, reject) => {
      const maybePromise = gutsOfYourAsyncFunc()

      if (isPromise(maybePromise)) {
          maybePromise.then(resolve).catch(reject)
     } else {
          resolve(maybePromise)
     }
  })
}
```